src/script/lua_api/l_server.cpp

```diff
@@ -447,6 +448,28 @@ int ModApiServer::l_get_worldpath(lua_State *L)
 	return 1;
 }
 
+// get_mod_data_path()
+int ModApiServer::l_get_mod_data_path(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	lua_rawgeti(L, LUA_REGISTRYINDEX, CUSTOM_RIDX_CURRENT_MOD_NAME);
+	if (!lua_isstring(L, -1)) {
+		return 0;
+	}
+
+	std::string modname = readParam<std::string>(L, -1);
+
+	const Server *srv = getServer(L);
+	std::string path = srv->getModDataPath() + DIR_DELIM + modname;
```

Comment:
There can be two different installed mods with the same name. And the data format they use can be incompatible.

Example: Mod foo/bar uses format A. Then some fork baz/bar (used in another world) reads the data (in format A) and updates it to its new format B. The foo/bar mod from the first world then can no longer read its data.

I suggest generating a name from the modpath (like the paths in world.mt). I.e. replace `/` with `.` and escape other `.`s.

Reply:
This would lose the mod data if you move it from worldmods to mods, or to a game for some reason.
IMO we should ignore this edge case. (Consider also how `secure.http_mods` grants it to any mod of that name, no matter where it's installed.)